test/martha/martha_v3.test.js

```diff
@@ -728,6 +724,83 @@ test.serial('martha_v3 parses HCA response correctly', async (t) => {
     sinon.assert.callCount(getJsonFromApiStub, 1);
 });
 
+const testWithTimeout = (ms, asyncTestFn) => (t) => {
+    const waitThenFail = async () => {
+        await delay(ms);
+        t.fail(`Test did not complete within ${ms}ms`);
+    };
+    return Promise.race([asyncTestFn(t), waitThenFail()]);
+};
+
+test.serial('martha_v3 succeeds within 60 seconds even if fetching a signed URL never returns', testWithTimeout(5 * 1000, async (t) => {
+    overridePencilsDownTime(3);
+    const {
+        id: objectId, self_uri: drsUri,
+        access_methods: { 0: { access_id: accessId } }
+    } = kidsFirstDrsResponse;
+    const bond = bondUrls('kids-first');
+    const drs = drsUrls(config.HOST_KIDS_FIRST_STAGING, objectId, accessId);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(kidsFirstDrsResponse);
+    getJsonFromApiStub.withArgs(bond.accessTokenUrl, terraAuth).resolves(bondAccessTokenResponse);
+    getJsonFromApiStub.withArgs(drs.accessUrl, `Bearer ${bondAccessTokenResponse.token}`).callsFake(async () => {
+        await delay(90 * 1000);
+        return 'Boom!';
+    });
+    const response = mockResponse();
+
+    await marthaV3(mockRequest({ body: { 'url': drsUri } }), response);
+
+    t.is(response.statusCode, 200);
+    t.deepEqual(response.body, kidsFirstDrsMarthaResult(null));
+    sinon.assert.callCount(getJsonFromApiStub, 3);
+}));
+
+test.serial('martha_v3 fails if something times out before trying to fetch a signed URL', testWithTimeout(5 * 1000, async (t) => {
+    overridePencilsDownTime(3);
+    const { id: objectId, self_uri: drsUri } = kidsFirstDrsResponse;
+    const drs = drsUrls(config.HOST_KIDS_FIRST_STAGING, objectId);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).callsFake(async () => {
+        await delay(90 * 1000);
+        return 'Boom!';
+    });
+    const response = mockResponse();
+
+    await marthaV3(mockRequest({ body: { 'url': drsUri } }), response);
+
+    t.is(response.statusCode, 500);
+    t.deepEqual(
+        response.body,
+        {
+            response: {
+                status: 500,
+                text: 'Timed out resolving DRS URI. Could not fetch DRS metadata.',
+            },
+            status: 500,
+        },
+    );
+}));
+
+test.serial('martha_v3 succeeds if an error is encountered while fetching a signed URL', async (t) => {
+    const {
+        id: objectId, self_uri: drsUri,
+        access_methods: { 0: { access_id: accessId } }
+    } = kidsFirstDrsResponse;
+    const bond = bondUrls('kids-first');
+    const drs = drsUrls(config.HOST_KIDS_FIRST_STAGING, objectId, accessId);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(kidsFirstDrsResponse);
+    getJsonFromApiStub.withArgs(bond.accessTokenUrl, terraAuth).resolves(bondAccessTokenResponse);
+    getJsonFromApiStub.withArgs(drs.accessUrl, `Bearer ${bondAccessTokenResponse.token}`)
```

Comment:
I discovered that this test is finicky in that it will still pass even if this stub doesn't get invoked. (I discovered this because I accidentally put a `:` after `Bearer` at first and the test didn't fail when I expected it to. Yay TDD!) I'm going to think about it a little more, but I think I'm going to have to add some assertions that the stub was actually invoked... much like the checks I removed during the test refactoring.